src/index.ts

```diff
@@ -1323,6 +1323,55 @@ async function claimLink({
 	}
 }
 
+/**
+ * Prepares a transaction to claim a link. Broadcasting the transaction is not handled by this function.
+ * @param link: the link to claim
+ * @param recipientAddress: the address to send the link's contents
+ * @param provider: the provider to use for the transaction
+ * @returns
+ */
+async function prepareClaimTx({
+	link,
+	recipientAddress,
+	provider = undefined,
+}: {
+	recipientAddress: string
+	link: string
+	provider?: ethers.providers.JsonRpcProvider
```

Comment:
Optional Provider object is how we currently do things, but if an integrator doesn't use ethers, but has private Providers, there should be a way of inputting those (so, an rpc URL string)

Do you know if there's any other place in the code base where we take in an rpc url string? I think there was one place but not sure

Reply:
Yes, I agree, but dont think we should change this only here. We are using jsonRpcProvider throughout the entire sdk, so only changing this here is a small change. I think if we want to do this properly, we should do it throughout the entire sdk. I've added in todo comment to remind us in the future. Sound good?